design/sds-design.md

```diff
@@ -0,0 +1,68 @@
+# Support Secret Discovery Service on Contour
+
+_Status_: Draft
+
+This document outlines what changes are needed on Contour to support SDS. [0]
+
+## Goals
+
+- Implement SDS server as a gRPS service
+- Support auth.SdsSecretConfig configuration on Listener & Cluster under auth.CommonTlsContext
```

Comment:
No matter how big the scope is (for the question above), the change mentioned above most likely is what we need, as an initial step, for enabling SDS. And that change can be addressed first no matter we decide to bake SDS server into Contour or not.

It's going to be a change somehow similar to the below one on `Istio` side

- https://github.com/istio/api/pull/547
- https://github.com/istio/istio/pull/6453

@bmadhavan @davecheney Do we agree on that?

Reply:
True. This change of supporting `auth.SdsSecretConfig configuration on Listener & Cluster` is a task that has to be done irrespective of which SDS server to use. I can submit a PR for those changes first.